lib/hsail-amdgpu-wrapper.ll

```diff
@@ -393,170 +388,155 @@ define linkonce_odr spir_func float @__hsail_ftz_f32(float) #1 {
 }
 
-; Function Attrs: alwaysinline
+; Function Attrs: nounwind readnone
 define linkonce_odr spir_func i32 @__hsail_get_local_id(i32) #1 {
-  %2 = call i32 @llvm.r600.read.tidig.x()
-  %3 = call i32 @llvm.r600.read.tidig.y()
-  %4 = call i32 @llvm.r600.read.tidig.z()
+  %2 = call i32 @llvm.amdgcn.workitem.id.x(), !range !1
+  %3 = call i32 @llvm.amdgcn.workitem.id.y(), !range !1
+  %4 = call i32 @llvm.amdgcn.workitem.id.z(), !range !1
   %5 = icmp eq i32 %0, 1
   %6 = select i1 %5, i32 %3, i32 %2
   %7 = icmp eq i32 %0, 2
   %8 = select i1 %7, i32 %4, i32 %6
   ret i32 %8
 }
 
-; Function Attrs: alwaysinline
+; Function Attrs: nounwind readnone
 define linkonce_odr spir_func i32 @__hsail_get_group_id(i32) #1 {
-  %2 = call i32 @llvm.r600.read.tgid.x()
-  %3 = call i32 @llvm.r600.read.tgid.y()
-  %4 = call i32 @llvm.r600.read.tgid.z()
+  %2 = call i32 @llvm.amdgcn.workgroup.id.x()
+  %3 = call i32 @llvm.amdgcn.workgroup.id.y()
+  %4 = call i32 @llvm.amdgcn.workgroup.id.z()
   %5 = icmp eq i32 %0, 1
   %6 = select i1 %5, i32 %3, i32 %2
   %7 = icmp eq i32 %0, 2
   %8 = select i1 %7, i32 %4, i32 %6
   ret i32 %8
 }
 
-; Function Attrs: alwaysinline
+; Function Attrs: alwaysinline nounwind readonly
 define linkonce_odr spir_func i32 @__hsail_currentworkgroup_size(i32) #0 {
   %dispatch_ptr = call noalias i8 addrspace(2)* @llvm.amdgcn.dispatch.ptr()
-  %dispatch_ptr_i32 = bitcast i8 addrspace(2)* %dispatch_ptr to i32 addrspace(2)*
-  %size_xy_ptr = getelementptr i32, i32 addrspace(2)* %dispatch_ptr_i32, i32 1
-  %size_xy = load i32, i32 addrspace(2)* %size_xy_ptr, align 4, !invariant.load !0
-  switch i32 %0, label %8 [
-    i32 0, label %2
-    i32 1, label %4
-    i32 2, label %6
+  %size_xy_ptr = getelementptr i8, i8 addrspace(2)* %dispatch_ptr, i64 4
+  %2 = bitcast i8 addrspace(2)* %size_xy_ptr to i32 addrspace(2)*
+  %size_xy = load i32, i32 addrspace(2)* %2, align 4, !invariant.load !0
+  switch i32 %0, label %10 [
+    i32 0, label %3
+    i32 1, label %5
+    i32 2, label %7
   ]
 
-; <label>:2                                       ; preds = %1
-  %3 = and i32 %size_xy, 65535 ; 0xffff
-  ret i32 %3
+; <label>:3:                                      ; preds = %1
+  %4 = and i32 %size_xy, 65535
+  ret i32 %4
 
-; <label>:4                                      ; preds = %1
-  %5 = lshr i32 %size_xy, 16
-  ret i32 %5
+; <label>:5:                                      ; preds = %1
+  %6 = lshr i32 %size_xy, 16
+  ret i32 %6
 
-; <label>:6                                      ; preds = %1
-  %size_z_ptr = getelementptr i32 ,i32 addrspace(2)* %dispatch_ptr_i32, i32 2
-  %7 = load i32, i32 addrspace(2)* %size_z_ptr, align 4, !invariant.load !0, !range !1
-  ret i32 %7
+; <label>:7:                                      ; preds = %1
+  %size_z_ptr = getelementptr i8, i8 addrspace(2)* %dispatch_ptr, i64 8
+  %8 = bitcast i8 addrspace(2)* %size_z_ptr to i32 addrspace(2)*
+  %9 = load i32, i32 addrspace(2)* %8, align 4, !range !2, !invariant.load !0
```

Comment:
@arsenm  There's no metadata !2 in the cleaned up version, could you take a look?